### PR TITLE
docs(dart): update BASICS intro

### PR DESCRIPTION
**Pull request by kwalrath**
Also copy edited the TS version a bit.

NOTE: The TS version (and thus the Dart version) talks about "Reference" and "Resources" sections that don't exist. (API Reference does, but not Reference, as such.)

### CODE REVIEW

**Comment by kwalrath** [comment]
@wardbell do you want me to make the section descriptions match what's really there?

**Comment by wardbell** [comment]
@kwalrath Please do :-)

File: public/docs/dart/latest/guide/index.jade [hunk 1]
```diff
@@ -1,93 +1,18 @@
...
+block example-links
+  :marked
+    Look for a link to that sample near the top of each page.
+    For example, the sample repo https://github.com/angular-examples/architecture
+    contains the code
+    for the [Architecture](architecture.html) chapter's sample.
+    A running version of that sample is at
+    https://angular-examples.github.io/architecture/.
+
+    A few early chapters are written as tutorials and are clearly marked as such.
```
Comment (chalin) [inline public/docs/dart/latest/guide/index.jade]: Why not include this paragraph (L15-18) in the TS version?

**Comment by chalin** [comment]
Other than the one inlined comment made about moving a paragraph to the TS version, it LGTM.

File: public/docs/dart/latest/guide/index.jade (hunk 1 above)
Comment (kwalrath) [inline public/docs/dart/latest/guide/index.jade]: It's the "live link" vs. "full repo" difference. I could've made it a variable and probably should.

Comment (chalin) [inline public/docs/dart/latest/guide/index.jade]: Ah, right. I just feel that we should minimize differences between the language versions. The added remark will be useful to TS and JS users too :). If you don't want to define a variable, you could just drop the last sentence (L18) and put L15-17 in the TS version. It's up to you.

Comment (kwalrath) [inline public/docs/dart/latest/guide/index.jade]: It actually _is_ in the TS & JS pages, but they have an all-plunkr solution where they can change the example. So the text needs to be different.

**Comment by kwalrath** [comment]
About to merge...

**Pull request merged**